File: DigitalLibrary.Models/Parameters/BookParameters.cs
using System;
using System.Collections.Generic;

namespace DigitalLibrary.Models.Parameters
{
    public class BookParameters : QueryStringParameters
    {
        public BookParameters()
        {
            OrderBy = "Title";
        }

        public string LibraryId { get; set; } = null;

        public bool OnlyInStorage { get; set; } = false;
        public int MinYear { get; set; } = 0;
        public int MaxYear { get; set; } = DateTime.Now.Year;

        public string SearchField { get; set; } = "Title";
        public string SearchTerm { get; set; }

        public int MinPages { get; set; } = 0;
        public int MaxPages { get; set; } = 0;

        public IEnumerable<string> GenreIds { get; set; } = new List<string>();
        public IEnumerable<string> SubjectIds { get; set; } = new List<string>();
    }
}

File: DigitalLibrary.Client/NGClient/src/app/shared/parameters/book-parameters.ts

export class BookParameters {
        pageSize: number = 4; 
        searchTerm?: string; 
        minYear?:number;
        maxYear?:number; 
        minPages?:number; 
        maxPages?:number; 
        onlyInStorage?:boolean;
        orderBy?:string; 
        searchField?: string;
        genreIds?: string[];
        subjectIds?: string[]; 
        pageNumber?: number;
        libraryId?: string;

        constructor(pageSize?, genreIds?, subjectIds?, searchTerm?, minYear?, maxYear?, minPages?, maxPages?, onlyInStorage?, orderBy?, searchField?) {
                this.pageSize = pageSize,
                this.genreIds = genreIds,
                this.subjectIds = subjectIds,
                this.searchTerm = searchTerm,
                this.minYear = minYear,
                this.maxYear = maxYear,
                this.minPages = minPages,
                this.maxPages = maxPages,
                this.onlyInStorage = onlyInStorage,
                this.orderBy = orderBy,
                this.searchField = searchField
        }
}